Pollo/ViewControllers/CardController+Extension.swift

```diff
@@ -220,27 +221,43 @@ extension CardController: UIScrollViewDelegate {
 
 extension CardController: SocketDelegate {
 
-    func sessionConnected() {}
+    func sessionConnected() {
+        print("Connected to socket")
+        let banner = NotificationBanner.connectedBanner()
+        self.currentBanner = banner
+        collectionView.isUserInteractionEnabled = true
+    }
     
-    func sessionDisconnected() {}
+    func sessionDisconnected() {
+        let banner = NotificationBanner.disconnectedBanner()
+        banner.onTap = { [weak self] in
+            guard let `self` = self else { return }
+            banner.dismiss()
+            self.socket.socket.setReconnecting(reason: "")
+        }
+
+        self.currentBanner = banner
+        collectionView.isUserInteractionEnabled = false
+    }
+
+    func sessionReconnecting(_ reason: Any?) {
+        let reason = reason as? String ?? ""
 
-    func sessionErrored() {
-        socket.socket.connect(timeoutAfter: 5) { [weak self] in
+        let banner = NotificationBanner.reconnectingBanner(reason: reason)
+        self.currentBanner = banner
+        self.collectionView.isUserInteractionEnabled = false
+
+        socket.socket.connect(timeoutAfter: 10) { [weak self] in
```

Comment:
socket.socket - any chance that we can rename one of these?

Reply:
we use it everywhere through the app so its kind of hard to rename it at this point tbh

Reply:
^^